### PR TITLE
Publish Reports in Jenkins

### PR DESCRIPTION
**Pull request by rajkatla-hmcts**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/RDCC-3640

### Change description ###
Publishing Test Reports in Jenkins

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

**Review by kavithagade12 — DISMISSED** [review]

**Review by amitkumarshrivastava — DISMISSED** [review]

File: Jenkinsfile_nightly [hunk 1]
```diff
@@ -81,6 +81,31 @@ withNightlyPipeline(type, product, component) {
...
+                keepAll              : true,
+                reportDir            : "build/reports/pitest",
+                reportFiles          : "index.html",
+                reportName           : "Mutation Test Report"
+        ]
+    }
+
+    after('fullFunctionalTest') {
+        steps.archiveArtifacts allowEmptyArchive: true, artifacts: '**/site/serenity/**/*'
+        steps.archiveArtifacts allowEmptyArchive: true, artifacts: 'build/reports/tests/**/*'
```
Comment (ManishParyani) [inline Jenkinsfile_nightly:98]: I am curious to know this. Do we need these steps? if yes then why are we not adding these steps to other test types i.e mutation test, smoke test.

File: Jenkinsfile_CNP [hunk 2]
```diff
@@ -62,23 +62,88 @@ withPipeline(type, product, component) {
...
+
+    after('pact-provider-verification') {
+        publishHTML target: [
+             allowMissing         : true,
+             alwaysLinkToLastBuild: true,
+             keepAll              : true,
+            reportDir            : "build/reports/tests/runProviderPactVerification",
+            reportFiles          : "index.html",
+            reportName           : "Contract Tests Report",
+            alwaysLinkToLastBuild: true
```
Comment (SujeethKumarAri) [inline Jenkinsfile_CNP:141]: you have already added this in line 136

Comment (rajkatla-hmcts) [reply Jenkinsfile_CNP]: removed duplicate line, thanks for spotting.

File: Jenkinsfile_nightly (hunk 1 above)
Comment (rajkatla-hmcts) [reply Jenkinsfile_nightly]: yes, we need these steps. 
First line is for serenity reports only and second line is to archive test reports at last stage, here functional test is last step in nightly job.

**Review by SujeethKumarAri — APPROVED** [review]

**Review by ishratt — APPROVED** [review]

**Pull request merged**